services/pull/update.go

```diff
@@ -37,7 +40,11 @@ func Update(pull *models.PullRequest, doer *models.User, message string) error {
 		return fmt.Errorf("HeadBranch of PR %d is up to date", pull.Index)
 	}
 
-	_, err = rawMerge(pr, doer, models.MergeStyleMerge, message)
+	if rebase {
+		err = doRebase(pr, doer)
+	} else {
+		_, err = rawMerge(pr, doer, models.MergeStyleMerge, message)
```

Comment:
I suppose I am misunderstanding something. 
But why can't we just change `models.MergeStyleMerge` to `models.MergeStyleRebase`.?

If there is a non-trivial reason, maybe add a brief 'why' comment here.

Reply:
See https://github.com/go-gitea/gitea/pull/16125#discussion_r649215822

In a words, ``MergeStyleRebase`` means rebase the head branch onto base branch, then merge it onto the base branch. But for ``update pull request by rebase``, just rebase head branch onto base branch is enough. Thanks.

Reply:
Though it's possible (and desirable) to unify the two paths, it's probably harder than I thought.

As far as I can tell, the difference mentioned in https://github.com/go-gitea/gitea/pull/16125#discussion_r649215822 can be addressed by 
1. Split `RawMerge` into two functions and reuse the first one here.
2. Add an extra parameter to `RawMerge` to turn off the second half of steps.